build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/docker/DockerBuildTask.java

```diff
@@ -181,7 +185,7 @@ public void execute() {
             }
 
             final List<String> tags = parameters.getTags().get();
-            final boolean isCrossPlatform = parameters.getPlatform().get().equals(Architecture.current().dockerPlatform) == false;
+            final boolean isCrossPlatform = isCrossPlatform();
```

Comment:
Seems unnecessary to have this field now that this logic lives in the `isCrosPlatform()` method.